File: ems-web/src/model/employee.ts
export type Employee = {
  id: number
  firstName: string
  lastName: string
  email: string
  jobTitle: string
  yearsOfService: number
  dateOfJoining: Date
}

export const employeeMockData: Employee[] = [
  {
    "id": 1,
    "firstName": "string",
    "lastName": "string",
    "email": "string",
    "jobTitle": "string",
    yearsOfService: 1,
    dateOfJoining: new Date()
  },
  {
    "id": 2,
    "firstName": "veeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeery",
    "lastName": "long naaaaaaaaaaame",
    "email": "string",
    "jobTitle": "string",
    yearsOfService: 1,
    dateOfJoining: new Date()
  }
]

File: EMS/EMS.API/Models/Employee.cs
namespace EMS.API.Models;

public class Employee
{
    public int ID { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Email { get; set; }
    public string JobTitle { get; set; }
    public DateTime DateOfJoining { get; set; }
    public int YearsOfService
    {
        get
        {
            return (int) (DateTime.UtcNow - DateOfJoining).TotalDays / 365;
        }
    }
}